### ISSUE TITLE
Trash bugs

### ISSUE THREAD

**Issue by taniarascia** [taniarascia/takenote#10]
- Add ability to permanently delete note (remove from `notes` array)
- Add ability to restore deleted note (remove `trash` property)
- Do not allow making a new note from trash (disable `addNote` if Folder is `TRASH`)

**Comment by yazeedb** [comment]
Hi! I'd be interested in trying that first bullet.

Can you permanently delete from other screens, or just _Trash_?

If just _Trash_, then `NoteOptions.tsx` may be extended easily? For example...

```tsx
{!clickedNote.trash && <PermanentlyDelete onClick={permanentlyDeleteHandler} /> }
```
After that, I think we just wire up `actions/index.ts` and `noteReducer.ts` to permanently delete.

**Comment by taniarascia** [comment]
I believe there’s still a DELETE_NOTE Action in the reducer that can be utilized. I think ideally it would only be delete from trash, and every other screen would just send to trash, so yes.

We’d probably want an “Empty trash” option as well.

I also like having a component like that, I’ll make a note to refactor other actions into components and maybe just send the Id as a prop.

**Comment by yazeedb** [comment]
What do you think of using a new name vs. reusing "Delete note"?

If something's already in the trash, _deleting_ it implies permanent erasure.

<img width="704" alt="Screen Shot 2019-10-12 at 3 55 33 PM" src="https://user-images.githubusercontent.com/6687119/66707038-c3158700-ed08-11e9-8565-59ab49bd0904.png">

**Comment by taniarascia** [comment]
I think no options from trash except permanently delete and restore. Since both of those options don’t exist on NoteOptions, it makes sense to be a totally separate component. 

Though it would be nice to make all the other code about opening and closing options be the same, so only the content of the options dropdown changes, and the functionality is reusable.

**Comment by yazeedb** [comment]
> ...it makes sense to be a totally separate component

Agreed.

> Though it would be nice to make all the other code about opening and closing options be the same...

Still reading up, but are you referring to stuff like `handleNoteOptionsClick`?

Perhaps a `NoteOptionsContainer` can interact with all that and receive a child `NoteOptions` or `TrashedNoteOptions` based on `note.trash`.

**Comment by yazeedb** [comment]
Just dug more into this, may not be that simple lol.

`NoteOptions` must `connect` to all the categories logic and be aware of the active `noteOptionsId`.

Then `TrashedNoteOptions` `connect`s to whatever it needs to delete and restore items.

Two Redux-connected components, then `NoteList` becomes considerably slimmer as it simply decides which one to render based on `note.trash`.

**Comment by taniarascia** [comment]
Yes, this works. I’d still like to separate the click and close logic out to prevent duplicate code, but that can be another issue.

**Comment by taniarascia** [comment]
Restore implemented in #27
